header/Utility/TextSplitter.h

```diff
@@ -0,0 +1,25 @@
+//
+// Created by dawid on 23.10.2020.
+//
+
+#ifndef GPU_KARP_RABIN_TEXTSPLITTER_H
+#define GPU_KARP_RABIN_TEXTSPLITTER_H
+
+#include <cmath>
+#include <tuple>
+#include <vector>
+
+class TextSplitter {
+
+private:
+    std::vector<std::tuple<long long int, long long int>> ranges;
```

Comment:
So actually the types stored here do not require to be `long long int`. It's better to use type `size_t` as it's type for storing index values. It is guaranted to have range as big as the largest memory block that you can allocate on the device. It's an unsigned type.